core/src/main/java/org/mule/runtime/core/internal/config/CustomServiceRegistry.java

```diff
@@ -27,4 +28,15 @@ public interface CustomServiceRegistry extends CustomizationService {
    * @return the registered custom services. Non null.
    */
   Map<String, CustomService> getCustomServices();
+
+  /**
+   * Overrides the given service implementation if a {@link ServiceOverrider} has been registered for the given {@code serviceId}.
+   *
+   * @param serviceId   identifier of the service.
+   * @param serviceImpl the service implementation to be overridden.
+   * @return the overridden service implementation, or {@link Optional#empty()} if it should be ignored.
+   * @since 4.6
+   */
+  Optional<Object> overrideDefaultService(String serviceId, Object serviceImpl);
```

Comment:
why does this CustomServiecRegistry needs to exist? WHy does something callled registry extend CustomizationService?